packages/js/e2e-core-tests/src/specs/merchant/wp-admin-order-edit.test.js

```diff
@@ -69,14 +77,152 @@ const runEditOrderTest = () => {
+				productId,
+				customerBilling ,
+				status: orderStatus.processing
+			} );
+		} );
+
+		it( 'can add downloadable product permissions to order without product', async () => {
+			// Create order without product
+			orderId = await createOrder( {
+				customerBilling,
+				status: orderStatus.processing
+			} );
+
+			// Open order we created
+			await merchant.goToOrder( orderId );
+
+			// Add permission
+			await merchant.addDownloadableProductPermission( productName );
+
+			// Verify new downloadable product permission details
+			await merchant.verifyDownloadableProductPermission( productName )
+		} );
+
+		it( 'can add downloadable product permissions to order with product', async () => {
+			// Create new downloadable product
+			const product2Name = 'TDP 002';
+			productId = await createSimpleDownloadableProduct( product2Name );
+
+			// Open order we created
+			await merchant.goToOrder( orderId );
+
+			// Add permission
+			await merchant.addDownloadableProductPermission( product2Name );
+
+			// Verify new downloadable product permission details
+			await merchant.verifyDownloadableProductPermission( product2Name )
+		} );
+
+		it( 'can edit downloadable product permissions', async () => {
+			// Define expected downloadable product attributes
+			const expectedDownloadsRemaining = '10';
+			const expectedDownloadsExpirationDate = '2050-01-01';
+
+			// Open order we created
+			await merchant.goToOrder( orderId );
+
+			// Update permission
+			await merchant.updateDownloadableProductPermission(
+				productName,
+				expectedDownloadsExpirationDate,
+				expectedDownloadsRemaining
+			);
+
+			// Verify new downloadable product permission details
+			await merchant.verifyDownloadableProductPermission(
+				productName,
+				expectedDownloadsExpirationDate,
+				expectedDownloadsRemaining
+			);
+		} );
+
+		it( 'can revoke downloadable product permissions', async () => {
+			// Open order we created
+			await merchant.goToOrder( orderId );
+
+			// Revoke permission
+			await merchant.revokeDownloadableProductPermission( productName );
+
+			// Verify
+			await expect( page ).not.toMatchElement( 'div.order_download_permissions', {
+				text: productName
+			} );
+		} );
+
+		it( 'should not allow downloading a product if download attempts are exceeded', async () => {
+			// Define expected download error reason
+			const expectedReason = 'Sorry, you have reached your download limit for this file';
+
+			// Setup data
+			productId = await createSimpleDownloadableProduct( productName, 0 );
+			orderId = await createOrder( {
+				productId,
+				customerBilling,
+				status: orderStatus.processing
+			} );
+
+			// Open order we created
+			await merchant.goToOrder( orderId );
+
+			// Open download page
+			const downloadPage = await merchant.openDownloadLink();
+
+			// Verify file download cannot start
+	  		await merchant.verifyCannotDownloadFromBecause( downloadPage, expectedReason );
+		} );
```

Comment:
In headless mode, this test (`should not allow downloading a product if download attempts are exceeded`) and the next test (`should not allow downloading a product if expiration date is exceeded`) passes on my end. 

However, when I ran these two tests using `it.only()` in UI mode, the  `download attempts are exceeded` test got stuck on this part, and did not proceed to the `expiration date is exceeded` test:
<img width="1523" alt="Screen Shot 2022-01-21 at 10 29 42 PM" src="https://user-images.githubusercontent.com/4509348/150547665-8567271b-e3dc-4fac-811d-6738a9bc7d97.png">

Do you also experience the same behavior when running these two tests in `test:e2e-dev` mode?

I have a feeling that this is because the `WordPress > Error` tab that was switched to [here](https://github.com/woocommerce/woocommerce/pull/31650/files#diff-131231274174823411e708b8712c4a010dfb09122a4bc6e4fc7a8a45ec443c2dR288) was not closed after verifying the error message.

Maybe a combination of `browser.pages()` and `page.close()` could help, but I'm not 100% sure as I haven't personally encountered the need for them so far.

Reply:
Oh nice one, I didn't spot that, thanks! I could reproduce it on my end.

```
await merchant.verifyCannotDownloadFromBecause( downloadPage, expectedReason );
```

We are passing the download page to that method, so we just need to close the page inside and that would solve the problem:
```
verifyCannotDownloadFromBecause: async ( page, reason ) => {
	// Select download page tab
	await page.bringToFront();

	// Verify error in download page
	await expect( page.title() ).resolves.toMatch( 'WordPress › Error' );
	await expect( page ).toMatchElement( 'div.wp-die-message', {
		text: reason
	} );

	// Close tab
	await page.close();
}
```
I checked it in both headless and browser modes and it passed.

Thanks again!

Reply:
> We are passing the download page to that function, so we just need to close the page inside
Ah right! Yep I believe that would solve it 👍